python/lsst/ip/isr/isrTask.py

```diff
@@ -831,9 +838,28 @@ def runQuantum(self, butlerQC, inputRefs, outputRefs):
                 if not isinstance(inputs["defects"], Defects):
                     inputs["defects"] = Defects.fromTable(inputs["defects"])
 
-        # TODO: DM-22776 add retrieval for brighter-fatter kernel for Gen3
-        # if we can get HSC to use a new kernel, or just translate the old
-        # one to the new format we could drop the whole new/old style support
+        # Load the correct style of brighter fatter kernel, and repack
+        # the information as a numpy array.
+        if self.config.doBrighterFatter:
+            brighterFatterKernel = inputs.pop('newBFKernel', None)
+            if brighterFatterKernel is None:
```

Comment:
why do this line, the if statement below will never be triggered by the bfKernel as it is an array

Reply:
If a newBFKernel exists in the inputs, then we should use that (and do the block beginning on line 848 to pull the kernel for the detector we have).  Otherwise, we should use the old style, which does not need to do the L848 block (and inputs['bfKernel'] remains what we received from the butler).